components/brave_wallet_ui/common/hardware/ledgerjs/fil_ledger_bridge_keyring.test.ts

```diff
@@ -0,0 +1,227 @@
+/* Copyright (c) 2022 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+ import { LEDGER_HARDWARE_VENDOR } from 'gen/brave/components/brave_wallet/common/brave_wallet.mojom.m.js'
+import { TextDecoder, TextEncoder } from 'util'
+import { BraveWallet } from '../../../constants/types'
+import { SignHardwareOperationResult } from '../types'
+import { FilGetAccountResponse, FilLotusMessage, FilSignedLotusMessage, FilSignTransactionResponse } from './fil-ledger-messages'
+import FilecoinLedgerBridgeKeyring from './fil_ledger_bridge_keyring'
+import { LedgerCommand, UnlockResponse } from './ledger-messages'
+import { MockLedgerTransport } from './ledger_bridge_keyring.test'
+global.TextDecoder = TextDecoder
+global.TextEncoder = TextEncoder
+
+// To use the MockLedgerTransport, we must overwrite
+// the protected `transport` attribute, which yields a typescript
+// error unless we use bracket notation, i.e. keyring['transport']
+// instead of keyring.transport. As a result we silent the dot-notation
+// tslint rule for the file.
+//
+/* eslint-disable @typescript-eslint/dot-notation */
+
+const unlockSuccessResponse: UnlockResponse = {
+  id: LedgerCommand.Unlock,
+  origin: window.origin,
+  command: LedgerCommand.Unlock,
+  payload: { success: true }
+}
+
+const unlockErrorResponse: UnlockResponse = {
+  id: LedgerCommand.Unlock,
+  origin: window.origin,
+  command: LedgerCommand.Unlock,
+  payload: {
+    success: false,
+    message: 'LedgerError',
+    statusCode: 101
+  }
+}
+
+const lotusMessage: FilLotusMessage = {
+  To: 'to',
+  From: 'from',
+  Nonce: 0,
+  Value: 'value',
+  GasPremium: 'gas',
+  GasLimit: 1,
+  GasFeeCap: 'cap',
+  Method: 1,
+  Params: 'params'
+}
+
+const signedLotusMessage: FilSignedLotusMessage = {
+  Message: lotusMessage,
+  Signature: {
+    Type: 1,
+    Data: 'data'
+  }
+}
+
+const getAccountsResponse: FilGetAccountResponse = {
+  payload: {
+    success: true,
+    accounts: ['0'],
+    deviceId: 'device1'
+  },
+  command: LedgerCommand.GetAccount,
+  id: LedgerCommand.GetAccount,
+  origin: 'origin'
+}
+
+const createKeyring = (): FilecoinLedgerBridgeKeyring => {
+  const ledgerHardwareKeyring = new FilecoinLedgerBridgeKeyring()
+  const transport = new MockLedgerTransport(window, window.origin)
+  ledgerHardwareKeyring['transport'] = transport
+  const iframe = document.createElement('iframe')
```

Comment:
please add the `sandbox` attribute: https://developer.mozilla.org/en-US/docs/Web/HTML/Element/iframe#attr-sandbox (and `allow-` values as needed)

Reply:
nvm i see this is just for testing, but might be good if it matched the actual iframe attributes: https://github.com/brave/brave-core/pull/14373/files#diff-2b646ba1b749f4939c65fc79a1e1a0aa8b85fdc9b5277376bb49b95be8c6a8ddR88